main/src/ca/uwaterloo/flix/api/lsp/LanguageServer.scala

```diff
@@ -98,6 +100,38 @@ class LanguageServer(port: Int, o: Options) extends WebSocketServer(new InetSock
     */
   private var currentErrors: List[CompilationMessage] = Nil
 
+  /**
+    * The java classes in the jdk classList
+    */
+  private val javaClasses: MultiMap[List[String], String] = sys.env.get("JAVA_HOME") match {
+    case None => MultiMap.empty
+    case Some(home) =>
+      val path = s"$home/lib/classlist"
+      try {
+        val file = scala.io.Source.fromFile(path)
+        val map = file.getLines()
+          // Filter the classlist file
+          .drop(5)
+          .filter(clazz => !clazz.contains("$"))
+          .filter(clazz => !clazz.contains("@"))
+          .foldLeft[MultiMap[List[String], String]](MultiMap.empty){
+            case (acc, clazz) =>
+              val clazzPath = clazz.split('/').toList
+              // Create a mapping from all prefixes to the next packages/classes
+              (0 until clazzPath.length).foldLeft(acc){
+                case (acc, prefixIdx) =>
+                  acc + (clazzPath.slice(0, prefixIdx) -> clazzPath(prefixIdx))
+              }
+          }
+        file.close
+        map
+      }
+      catch {
+        // If the users JDK does not include a classlist file or scala cannot find the file no map is created.
+        case _: FileNotFoundException => MultiMap.empty
+      }
+  }
+
   /**
```

Comment:
A question is: What happens if classlist is not part of the JDK in use or has a different format?